File: _includes/code/typescript/quickstart.short.query.rag.ts
// RAG
import weaviate, { WeaviateClient, ApiKey, generativeParameters } from 'weaviate-client';

// Best practice: store your credentials in environment variables
const weaviateUrl = process.env.WEAVIATE_URL!;
const weaviateApiKey = process.env.WEAVIATE_API_KEY!;
const anthropicApiKey = process.env.ANTHROPIC_API_KEY!;

// Step 2.1: Connect to your Weaviate Cloud instance
const client: WeaviateClient = await weaviate.connectToWeaviateCloud(
  weaviateUrl,
  {
    authCredentials: new ApiKey(weaviateApiKey),
    // highlight-start
    headers: { 'X-Anthropic-Api-Key': anthropicApiKey },
    // highlight-end
  }
);

// Step 2.2: Use this collection
const movies = client.collections.get('Movie');

// Step 2.3: Perform RAG with on NearText results
// highlight-start
const response = await movies.generate.nearText(
  'sci-fi',
  {
    groupedTask: 'Write a tweet with emojis about this movie.',
    config: generativeParameters.anthropic({
      model: "claude-3-5-haiku-latest",
    }),
  },
  {
    limit: 1,
  }
);
// highlight-end

console.log(response.generative); // Inspect the results

await client.close(); // Free up resources
// END RAG

File: _includes/code/python/quickstart.short.query.rag.py
# START RAG
import os
import weaviate
from weaviate.classes.generate import GenerativeConfig

# Best practice: store your credentials in environment variables
weaviate_url = os.environ["WEAVIATE_URL"]
weaviate_api_key = os.environ["WEAVIATE_API_KEY"]
anthropic_api_key = os.environ["ANTHROPIC_API_KEY"]

# Step 2.1: Connect to your Weaviate Cloud instance
client = weaviate.connect_to_weaviate_cloud(
    cluster_url=weaviate_url,
    auth_credentials=weaviate_api_key,
    # highlight-start
    headers={"X-Anthropic-Api-Key": anthropic_api_key},
    # highlight-end
)

# Step 2.2: Use this collection
movies = client.collections.use("Movie")

# Step 2.3: Perform RAG with on NearText results
# highlight-start
response = movies.generate.near_text(
    query="sci-fi",
    limit=1,
    grouped_task="Write a tweet with emojis about this movie.",
    generative_provider=GenerativeConfig.anthropic(),  # Configure the Anthropic generative integration for RAG
)
# highlight-end

print(response.generative.text)  # Inspect the results

client.close()  # Free up resources
# END RAG
